tests/snapshots.rs

```diff
@@ -233,3 +233,54 @@ fn convert_spv(name: &str) {
 fn convert_spv_shadow() {
     convert_spv("shadow");
 }
+
+#[cfg(feature = "glsl-in")]
+fn convert_glsl(name: &str, entry_points: Vec<(String, naga::ShaderStage)>, language: Language) {
+    let params = match std::fs::read_to_string(format!("tests/in/{}{}", name, ".param.ron")) {
+        Ok(string) => ron::de::from_str(&string).expect("Couldn't find param file"),
+        Err(_) => Parameters::default(),
+    };
+
+    let module = naga::front::glsl::parse_str(
+        &std::fs::read_to_string(format!("tests/in/{}{}", name, ".glsl"))
+            .expect("Couldn't find glsl file"),
+        entry_points,
+        Default::default(),
+    )
+    .unwrap();
+    naga::proc::Validator::new().validate(&module).unwrap();
+
+    #[cfg(feature = "spv-out")]
+    {
+        if language.contains(Language::SPIRV) {
+            check_output_spv(&module, name, &params);
```

Comment:
Here is a controversial proposal for you: let's make WGSL a "blessed" frontend.
The problem here is that, if we are going to have this sort of snapshot testing for all frontends, then we'll have way too much backend coverage. Its not bad in itself, but the dis-balance between frontend and backend coverage is bad.
So we need to be more selective. Most frontend testing should simply be `front -> IR` snapshots, like the current SPIR-V frontend snapshot. Only a few would need to go into all backends. We could select them arbitrarily, or we can say that all WGSL tests go this way. I think it's a fine direction because of the following reasons:
  1. WGSL is super close to our IR, so we aren't going to get into a situation where the backends only test specific subset of IR that the blessed frontend produces.
  2. it's consistent. We aren't going to ask ourselves this question in the future if we just make this call now.

Reply:
Heh, was actually thinking along those lines as well :)

Initially using the full `spv-out > spirv-val`, to make sure everything is good.

Reply:
Hmm after thinking some more: as glsl -> spv is likely to be a main use case (at least until wgsl takes over the world), so I'd like to at least have the option to do the full glsl -> spirv-val in our auto tests. Would also allow things like refactoring glsl-in to be done with more confidence :)

Anyway lemme know what you think.

Reply:
ok, sounds reasonable. So let's have GLSL->IR->SPV snapshots

Reply:
Ok, for now made glsl-in tests always do IR and spv